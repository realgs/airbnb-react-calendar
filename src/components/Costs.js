import React from 'react';

const Costs = ({ price, currency, stayLength, cleaningFee, bonus, amount, serviceFee, total }) => {

  return (
    <div className="costs">
      <div className="costs__cost">
        <span className="costs__cost__description">
          {price} x {stayLength} days
        </span>
        <span className="costs__cost__amount">{ `${currency}${amount}` }</span>
      </div>
      <div className="costs__cost">
        <span className="costs__cost__description">
          Cleaning fee
        </span>
        <span className="costs__cost__amount">{ `${currency}${cleaningFee}` }</span>
      </div>
      <div className="costs__cost">
        <span className="costs__cost__description">
          Service fee
        </span>
        <span className="costs__cost__amount">{`${currency}${serviceFee}`}</span>
      </div>
      { bonus > 0 && total > 0 && (
      <div className="costs__cost costs__bonus">
        <span className="costs__cost__description">
          Bonus funds
        </span>
        <span className="costs__cost__amount">{`-${currency}${bonus}`}</span>
      </div>
      )
      }
      <div className="costs__cost costs__sum">
        <span className="costs__cost__description">
          Total
        </span>
        <span className="costs__cost__amount">${ total >= 0 ? total : 0 }</span>
      </div>
    </div>
  );
};

export default Costs;