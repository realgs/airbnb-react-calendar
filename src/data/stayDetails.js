export const stayDetails = {
  user: 'Matthew.Jings',
  price: 19.9,
  rating : 3.5,
  reviews : 123,
  bonus : 10,
  cleaningFee : 10,
  stayLength : 0,
  minStay: 2,
};