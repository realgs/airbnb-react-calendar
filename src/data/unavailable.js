export const unavailableDates = [
  "2018-11-01",
  "2018-11-02",
  "2018-11-14",
  "2018-11-16",
  "2018-11-20",
  "2018-11-25",
  "2018-11-26",
  "2018-12-01",
  "2018-12-02",
  "2018-12-03",
  "2018-12-15",
  "2018-12-16",
  "2018-12-17",
  "2019-01-05",
  "2019-01-06",
  "2019-01-07",
  "2019-01-08",
  "2019-01-15",
  "2019-01-16",
  "2019-01-17",
  "2019-01-20",
  "2019-01-21",
  "2019-01-22",
  "2019-01-26"
];