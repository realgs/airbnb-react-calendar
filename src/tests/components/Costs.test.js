import React from 'react';
import { shallow } from 'enzyme';
import Costs from '../../components/Costs';
import { stayDetails } from '../../data/stayDetails';

const { price, currency, stayLength, cleaningFee, bonus } = { ...stayDetails };

test('Should match Costs snapshot.', () => {
  const wrapper = shallow(<Costs
    price={price}
    currency={currency}
    stayLength={stayLength}
    cleaningFee={cleaningFee}
    bonus={bonus}
  />);
  expect(wrapper).toMatchSnapshot();
});

test('Should not show bonus if <= 0', () => {
  const wrapper = shallow(<Costs bonus={0} />);
  expect(wrapper.find('.costs__bonus').exists()).toBe(false);
});

test('Should show bonus if both bonus and total > 0', () => {
  const wrapper = shallow(<Costs
    bonus={10}
    total={20}
  />);
  expect(wrapper.find('.costs__bonus').exists()).toBe(true);
});