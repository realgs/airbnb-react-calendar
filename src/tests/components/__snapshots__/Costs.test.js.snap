// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should match Costs snapshot. 1`] = `
<div
  className="costs"
>
  <div
    className="costs__cost"
  >
    <span
      className="costs__cost__description"
    >
      19.9
       x 
      0
       days
    </span>
    <span
      className="costs__cost__amount"
    >
      undefinedundefined
    </span>
  </div>
  <div
    className="costs__cost"
  >
    <span
      className="costs__cost__description"
    >
      Cleaning fee
    </span>
    <span
      className="costs__cost__amount"
    >
      undefined10
    </span>
  </div>
  <div
    className="costs__cost"
  >
    <span
      className="costs__cost__description"
    >
      Service fee
    </span>
    <span
      className="costs__cost__amount"
    >
      undefinedundefined
    </span>
  </div>
  <div
    className="costs__cost costs__sum"
  >
    <span
      className="costs__cost__description"
    >
      Total
    </span>
    <span
      className="costs__cost__amount"
    >
      $
      0
    </span>
  </div>
</div>
`;
